rust-runtime/aws-smithy-http-server/src/plugin/filter.rs

```diff
@@ -0,0 +1,42 @@
+use tower::util::Either;
+
+use crate::operation::{Operation, OperationShape};
+
+use super::Plugin;
+
+pub struct FilterByOperationName<Inner, F> {
+    inner: Inner,
+    predicate: F,
+}
+
+impl<Inner, F> FilterByOperationName<Inner, F> {
+    /// Creates a new [`FilterByOperationName`].
+    pub fn new(inner: Inner, predicate: F) -> Self {
```

Comment:
`pub(crate)` and force them to use the extension trait methods?

Reply:
I think that's good call 👌🏻